### ISSUE TITLE
Homeassistant not been updated when device become offline

### ISSUE THREAD

**Issue by grfernandes54** [zigbee2mqtt/hassio-zigbee2mqtt#237]
**Description of problem:**

I've tried to monitor when I had a power outage at my house using a zigbee socket by get the state of this socket every 10 seconds. The rule is if the socket reply with information is because it is online and have energy, if the socket did not reply is because I have a power outage

What is happening is, when I disconnect device from the energy, zigbee2mqtt stil passing to home assistant the last devices parameters and homeassistant cannot indentify that the device become offiline

print from zigbee2mqtt show the device expose after disconnected from the energy

<img width="386" alt="image" src="https://user-images.githubusercontent.com/70862164/142492596-bcf29d19-0020-44e0-acc7-695e36a2452d.png">

zigbee2mqttl log showing that device is offline

 Received MQTT message on 'zigbee2mqtt/Tomada Desumidificador/get' with data '{"state":""}'
Debug Publishing get 'get' 'state' to 'Tomada Desumidificador'
Debug Saving state to file /config/zigbee2mqtt/state.json
Error Publish 'get' 'state' to 'Tomada Desumidificador' failed: 'Error: Read 0x60a423fffeff8389/1 genOnOff(["onOff"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC no ack' (233))'
Debug Error: Read 0x60a423fffeff8389/1 genOnOff(["onOff"], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC no ack' (233)) at ZStackAdapter.sendZclFrameToEndpointInternal (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:390:23) at Queue.executeNext (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:32:32)
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'get' 'state' to 'Tomada Desumidificador' failed: 'Error: Read 0x60a423fffeff8389/1 genOnOff([\"onOff\"], {\"sendWhenActive\":false,\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":true,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (Data request failed with error: 'MAC no ack' (233))'","meta":{"friendly_name":"Tomada Desumidificador"},"type":"zigbee_publish_error"}'

print from HomeAssistant show that Zigbee2Mqtt keep reporting the previous devices parameters after disconnect it from the energy

<img width="504" alt="image" src="https://user-images.githubusercontent.com/70862164/142493007-d3990bf8-063b-4265-8a24-5036bb108bcf.png">

**Zigbee2mqtt add-on version (if edge, please report commit hash):** 1.22.0-2

Running platform (details from Suypervisor > System)** 
```
// Example
Core: core-2021.11.3
Supervisor: supervisor-2021.10.8
Operating System: Home Assistant OS 6.6
```

**Your logs from Home Assistant Supervisor**

21-11-18 16:55:43 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:0 is not WSMsgType.TEXT
21-11-18 16:55:43 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
21-11-18 16:55:54 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-11-18 16:55:54 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_nodered
21-11-18 16:55:54 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
21-11-18 17:08:57 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'igoradm'
21-11-18 17:08:57 INFO (MainThread) [supervisor.auth] Successful login for 'igoradm'
21-11-18 17:09:58 INFO (MainThread) [supervisor.host.info] Updating local host information
21-11-18 17:09:59 INFO (MainThread) [supervisor.host.services] Updating service information
21-11-18 17:09:59 INFO (MainThread) [supervisor.host.network] Updating local network information
21-11-18 17:10:00 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
21-11-18 17:10:01 INFO (MainThread) [supervisor.host.manager] Host information reload completed
21-11-18 17:12:38 INFO (MainThread) [supervisor.backups.manager] Creating new partial backup with slug 9616c801
21-11-18 17:12:38 INFO (MainThread) [supervisor.backups.manager] Backing up 9616c801 store Add-ons
21-11-18 17:12:38 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on a0d7b954_nodered
21-11-18 17:12:38 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon a0d7b954_nodered
21-11-18 17:12:38 INFO (MainThread) [supervisor.backups.manager] Creating partial backup with slug 9616c801 completed
21-11-18 17:12:39 INFO (SyncWorker_0) [supervisor.docker.interface] Updating image ghcr.io/hassio-addons/node-red/aarch64:10.1.0 to ghcr.io/hassio-addons/node-red/aarch64:10.1.1
21-11-18 17:12:39 INFO (SyncWorker_0) [supervisor.docker.interface] Downloading docker image ghcr.io/hassio-addons/node-red/aarch64 with tag 10.1.1.
21-11-18 17:12:45 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:45 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:45 INFO (MainThread) [supervisor.api.middleware.security] /backups/9616c801/info access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:46 INFO (MainThread) [supervisor.api.middleware.security] /backups/a7f2eda2 access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:47 INFO (MainThread) [supervisor.backups.manager] Removed backup file a7f2eda2
21-11-18 17:12:47 INFO (MainThread) [supervisor.api.middleware.security] /backups/9616c801/download access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:47 INFO (MainThread) [supervisor.api.backups] Downloading backup 9616c801
21-11-18 17:12:54 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:54 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:55 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
21-11-18 17:12:55 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup
21-11-18 17:14:04 INFO (SyncWorker_0) [supervisor.docker.interface] Stopping addon_a0d7b954_nodered application
21-11-18 17:14:06 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:0 is not WSMsgType.TEXT
21-11-18 17:14:06 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
21-11-18 17:14:07 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:09 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:11 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:13 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:15 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:16 INFO (SyncWorker_0) [supervisor.docker.interface] Cleaning addon_a0d7b954_nodered application
21-11-18 17:14:17 INFO (MainThread) [supervisor.addons] Add-on 'a0d7b954_nodered' successfully updated
21-11-18 17:14:17 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:17 INFO (SyncWorker_1) [supervisor.docker.interface] Cleanup images: ['ghcr.io/hassio-addons/node-red/aarch64:10.1.0']
21-11-18 17:14:19 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:21 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:23 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:25 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:27 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:28 INFO (SyncWorker_3) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/node-red/aarch64 with version 10.1.1
21-11-18 17:14:29 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:31 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:33 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:35 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:36 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:37 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:39 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:41 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:43 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:45 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:47 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:49 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:51 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:53 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:55 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:57 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:14:59 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:01 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:03 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:05 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:07 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:09 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:11 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:13 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:15 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:17 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:19 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:21 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:23 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.32.1:65288 ssl:default [Connect call failed ('172.30.32.1', 65288)]
21-11-18 17:15:27 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'igoradm'
21-11-18 17:15:27 INFO (MainThread) [supervisor.auth] Successful login for 'igoradm'
21-11-18 17:15:29 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-11-18 17:15:29 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_nodered
21-11-18 17:15:29 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
21-11-18 17:20:14 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
21-11-18 17:31:00 INFO (SyncWorker_0) [supervisor.docker.interface] Restarting ghcr.io/home-assistant/raspberrypi3-64-homeassistant
21-11-18 17:31:00 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Received message 8:1000 is not str
21-11-18 17:31:00 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
21-11-18 17:31:20 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
21-11-18 17:31:44 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
21-11-18 17:31:44 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-11-18 17:32:25 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-11-18 17:32:25 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_nodered
21-11-18 17:32:25 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
21-11-18 17:32:25 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running

**Comment by habitoti** [comment]
Just wondered about this, too. Being a deconz-migrant and MQTT newbie, I was still trying to figure out why my new setup doesn't propagate any offline state to HA. But so it seems this is not normal or just a mis-configuration?

**Comment by habitoti** [comment]
Seems to be fixed with 1.22.0-3 (after setting _availability_ to true)

**Comment by grfernandes54** [comment]
setting availability to true at zigbee2mqtt settings fixed the problem.

thank you @habitoti
